pkg/config/configparameters.go

```diff
@@ -69,3 +76,24 @@ func LoadParamFromJson(path string) (ConfigParameters, error) {
 
 	return tempParameters, nil
 }
+
+//parses the body of HTTP request (assumes json format, same as the config files), and returns the config struct
+func LoadParamFromHTTPRequest(r *http.Request) (ConfigParameters, error) {
+
+	var tempParameters ConfigParameters
+
+	err := json.NewDecoder(r.Body).Decode(&tempParameters)
+	if err != nil {
+		return tempParameters, err
+	}
+
+	//appending the sizes of the agents to the array
+	tempParameters.NumOfAgents = append(tempParameters.NumOfAgents, tempParameters.Team1Agents, tempParameters.Team2Agents, tempParameters.Team3Agents, tempParameters.Team4Agents, tempParameters.Team5Agents, tempParameters.Team6Agents, tempParameters.RandomAgents)
```

Comment:
This code is duplicated. Any way this can be reduced? (if it becomes too complicated passing things around, it's fine to have a duplication only)

Reply:
Thanks, that's a good point. I have now pushed a modified version where I added a function to reduce duplicated code.